### PR TITLE
fix: Fixing initialization & de-initialization issue on kits.

### PR DESCRIPTION
**Pull request by markvdouw**
Fixing initialization & de-initialization issue on kits.

## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Have and Android app with at least one integration kit connected and added as a dependency, re-run app after first install and you will see in the logs De-initializing kit: <kit name>. This happens quite often about 50% of the time.
Configuration was done pararelly for core & kits.
If any kit was present with a null or empty string (then converted to a JsonObject), the active kits were de-initialized and re-initialized because the preferences were overriden with empty string.

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - {explain how this has been tested, and what, if any, additional testing should be done}

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://mparticle-eng.atlassian.net/browse/SQDSDKS-5408

### CODE REVIEW

**Review by einsteinx2 — APPROVED** [review]
Approved per Slack discussion. Defer to Mo for final e2e testing before merging.

**Review by BrandonStalnaker — APPROVED** [review]

**Pull request merged**
